src/main/java/factionsplusplus/eventhandlers/JoinHandler.java

```diff
@@ -74,6 +78,24 @@ public void handle(PlayerJoinEvent event) {
         this.setPlayerActionBarTerritoryInfo(event.getPlayer());
         this.claimService.informPlayerIfTheirLandIsInDanger(player);
         this.informPlayerIfTheirFactionIsWeakened(player);
+        this.handleDynmapMapVisibility(player.getUniqueId());
+    }
+
+    private void handleDynmapMapVisibility(UUID uuid) {
+        Faction playerFaction = this.dataService.getPlayersFaction(uuid);
+
+        // In the event a player is hidden, leaves faction & rejoins, we should reset their visibility
+        if (playerFaction == null) {
```

Comment:
Could we not change this to:

if (playerFaction != null && !playerFaction.getEnemyFactions().isEmpty()) {
    this.dynmapIntegrationService.changePlayersVisibility(List.of(uuid), false);
    return;
}
this.dynmapIntegrationService.changePlayersVisibility(List.of(uuid), true);